.github/workflows/release.yml

```diff
@@ -49,16 +49,10 @@ jobs:
               { "id": "gpg.passphrase", "passphrase": "${{ secrets.DIKTAT_GPG_PASS }}" },
               { "id": "kotlin-maven-plugin-tools", "username": "${{github.actor}}", "password": "${{ secrets.GITHUB_TOKEN }}" }
             ]
-        # Runs tests, incl. smoke tests (using a fat JAR built from the
-        # "diktat-ruleset" module). The fat JAR is also used later during the
-        # "upload-release-asset" step (see below).
-      - name: Run tests
-        run: mvn -B clean install
-        # "diktat-ruleset" is a module that creates a fat JAR, which is,
-        # obviously, never released. Since we *do need* this fat JAR in order to
-        # run smoke tests, we *skip tests entirely* in this step.
-      - name: Deploy artifacts (skipping the clean, test, and verify phases)
-        run: mvn -B deploy -Prelease --projects "!diktat-ruleset" -DskipTests=true
+      - name: Deploy artifacts
+        run: mvn -B clean deploy -Prelease --projects '!diktat-ruleset'
+      - name: Build diktat.jar
+        run: mvn -B package --projects diktat-ruleset
```

Comment:
I thought that we would get rid of `diktat-ruleset` completely and upload fat jar from `diktat-rules`. What do you think about it?

Reply:
I've observed problems producing **two** JAR artifacts from a single module: `diktat-maven-plugin` fails to build for whatever reason, see https://github.com/saveourtool/diktat/blob/infra/andrey.shcheglov/misc/diktat-rules/pom.xml#L186.

Additionally, to preserve compatibility, for the fat JAR we need to provide a valid `groupId` and `artifactId` (different from those of `diktat-rules`), craft a custom `pom.xml`, etc. This is a way more general task than simply resolving a circular dependency b/w modules, with little profit at the end.